task_name: pr2_tasks/ChopStickTask

left_arm_ik_dmp_controller_name: l_arm_dmp_ik_controller
right_arm_ik_dmp_controller_name: r_arm_dmp_ik_controller
dual_arm_ik_dmp_controller_name: dual_arm_dmp_ik_controller

left_arm_joint_dmp_controller_name: l_arm_dmp_joint_position_controller
right_arm_joint_dmp_controller_name: r_arm_dmp_joint_position_controller
dual_arm_joint_dmp_controller_name: dual_arm_dmp_joint_position_controller

num_time_steps: 100

first_trial: 1
last_trial: 100

initialize_dmp_from_file: no
initial_dmp_file_name: /u/pastor/ros/pr2_tasks/library/initial_dmps/initial_chop_stick_dmp.bag

task_dmp_id: 7
reset_dmp_id: 8

task_movement_duration: 5.0
task_execution_duration: 6.0
reset_movement_duration: 8.0
reset_execution_duration: 8.0

dmp_sampling_frequency: 1000.0
root_frame: torso_lift_link
left_arm_tip_frame: l_gripper_tool_frame
right_arm_tip_frame: r_gripper_tool_frame

demonstration_bag_file: library/dmp_test_data/chop_stick_task_01.bag

control_cost_weight: 0.0
effort_cost_weight: 0.0
l_gripper_acceleration_weight: 0.0
r_gripper_acceleration_weight: 0.0
l_arm_effort_weight: 0.0
r_arm_effort_weight: 0.0
l_gripper_pressure_weight: 0.0
r_gripper_pressure_weight: 0.0
imu_weight: 1.0

# r_arm_effort_cost_joint_names: r_shoulder_pan_joint r_shoulder_lift_joint r_upper_arm_roll_joint r_elbow_flex_joint r_forearm_roll_joint r_wrist_flex_joint r_wrist_roll_joint
# l_arm_effort_cost_joint_names: l_shoulder_pan_joint l_shoulder_lift_joint l_upper_arm_roll_joint l_elbow_flex_joint l_forearm_roll_joint l_wrist_flex_joint l_wrist_roll_joint

num_rollouts: 8
num_reused_rollouts: 4

#              right_x right_y right_z  right_roll right_pitch right_yaw     right arm joint angles                   left_x left_y left_z  left_roll left_pitch left_yaw     left arm joint angles                   
noise_stddev: [0.0,    0.0,    0.0,     0.0,       0.0,        0.0,          0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0,       0.0,   0.0,   0.0,    0.0,      0.0,       0.0,         0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
noise_decay:  [0.99,   0.99,   0.99,    0.99,      0.99,       0.99,         0.99,0.99,0.99,0.99,0.99,0.99,0.99,      0.99,  0.99,  0.99,   0.99,     0.99,      0.99,        0.99,0.99,0.99,0.99,0.99,0.99,0.99]
